riscv_ctg/generator.py

```diff
@@ -892,6 +1006,11 @@ def swreg(self, instr_dict):
                         assigned += 1
                         if offset == 2048:
                             offset = 0
+                        if rd_is_paired:
```

Comment:
There is no need to handle having a separate case for when `rd` is paired. The same can be handled via the `stride` field in the `template.yaml` and using the `offset` appropriately in the assembly macro. This same concept can be applied in all the other places where size of the signature is needed i.e while calculating the signature region sizes.

Reply:
Fixed in 9615e820b03079ebfab3fce466b89646ab80ea60 followed by a fix to the swreg offset allocation algorithm in fa04c9bcb7cf09c144a135feb870dd301644f4ae